### ISSUE TITLE
Login issues using SOCKS5

### ISSUE THREAD

**Issue by NaughtDZ** [Nandaka/PixivUtil2#618]
### Prerequisites

* [n] Did you read FAQ section?
* [y] Did you test with the latest [releases](https://github.com/Nandaka/PixivUtil2/releases "latest releases") or [commit](https://github.com/Nandaka/PixivUtil2/commits/master "latest commit") ?

### Description
Run programe,and it said:

Donate at https://bit.ly/PixivUtilDonation
Reading G:\pixivutil\config.ini ...
done.
Using default DB Path: G:\pixivutil\db.sqlite
Creating database... done.
Only process members where the last update is >= 7 days ago
Using Username: **********
Trying to log in with saved cookie
Error at doLogin(): (<class 'mechanize._response.get_seek_wrapper_class.<locals>.httperror_seek_wrapper'>, <httperror_seek_wrapper (urllib.error.HTTPError instance) at 0xa0dc778 whose wrapped object = <HTTPError 302: 'The HTTP server returned a redirect error that would lead to an infinite loop.\nThe last 30x error message was:\nFound'>>, <traceback object at 0x0A0E42A8>)
Cannot Login!
press enter to exit.

--------------
well，I tryed direct connection with pixivDownloaser,BUt....

PixivDownloader2 version 20200103-beta2
https://github.com/Nandaka/PixivUtil2/releases
Donate at https://bit.ly/PixivUtilDonation
Reading G:\pixivutil\config.ini ...
done.
Using default DB Path: G:\pixivutil\db.sqlite
Creating database... done.
Only process members where the last update is >= 7 days ago
Using Username: 
Trying to log in with saved cookie
Error at doLogin(): (<class 'mechanize._response.get_seek_wrapper_class.<locals>.httperror_seek_wrapper'>, <httperror_seek_wrapper (urllib.error.HTTPError instance) at 0x98ca778 whose wrapped object = <HTTPError 302: 'The HTTP server returned a redirect error that would lead to an infinite loop.\nThe last 30x error message was:\nFound'>>, <traceback object at 0x098D4188>)
Cannot Login!
press enter to exit.

PS:I can open pixiv on chrome,And I tried reset netsock
-=-=-

**Comment by RecklessJames** [comment]
I am having this exact same issue as well, same error messages and I cannot login.

**Comment by Nandaka** [comment]
why you use proxifier? you can set the socks5 proxy in the config.ini.

**Comment by NaughtDZ** [comment]
> why you use proxifier? you can set the socks5 proxy in the config.ini.

As i said, write in config.ini is not working and still error message.

**Comment by RecklessJames** [comment]
Still getting the same issues here on my end as well.

**Comment by Nandaka** [comment]
are you running from source or from exe file?

**Comment by Nandaka** [comment]
Try [dist-20200108.zip](https://github.com/Nandaka/PixivUtil2/files/4034304/dist-20200108.zip)

Ensure config.ini set accordingly:
![image](https://user-images.githubusercontent.com/836867/71963674-1e3b1500-3237-11ea-9a93-5e60b886ba09.png)

**Comment by sl1pkn07** [comment]
same error without any socks5

~~~
└───╼  python PixivUtil2.py 
PixivDownloader2 version 20200112
https://github.com/Nandaka/PixivUtil2/releases
Donate at https://bit.ly/PixivUtilDonation
Reading /home/sl1pkn07/aplicaciones/PixivUtil2/config.ini ...
done.
Using default DB Path: /home/sl1pkn07/aplicaciones/PixivUtil2/db.sqlite 
Creating database... done.
Only process members where the last update is >= 7 days ago 
Using Username: sl1pkn07
Trying to log in with saved cookie 
Error at doLogin(): (<class 'mechanize._response.get_seek_wrapper_class.<locals>.httperror_seek_wrapper'>, <httperror_seek_wrapper (urllib.error.HTTPError instance) at 0x7f1368b6f640 whose wrapped object = <HTTPError 302: 'The HTTP server returned a redirect error that would lead to an infinite loop.\nThe last 30x error message was:\nFound'>>, <traceback object at 0x7f1368b74380>) 
Cannot Login! 
press enter to exit.
~~~

using python 3.8

**Comment by Nandaka** [comment]
using PySocks? cos it work when I test using local SOCKS5 and free SOCKS5

**Comment by sl1pkn07** [comment]
Pysocks is a dependency, if not, fail launch (missing module)

**Comment by Nandaka** [comment]
ya, I know, that's why I ask are you using PySocks? cos it is a replacement for SocksiPy-branch, which you need to remove from the pip list.

**Comment by sl1pkn07** [comment]
no, i'm not use any socks in the configuration

the python modules is installed througth distro package manager. the only SocksiPy related package i have installed is for python2 environment

**Comment by Nandaka** [comment]
Then I have no idea what happen to your enviroment as I'm using windows.

For reference: here are the packages that I'm install using pip.
```
Package             Version
------------------- ----------
apng                0.3.3
astroid             2.3.3
atomicwrites        1.3.0
attrs               19.3.0
beautifulsoup4      4.8.1
cachetools          4.0.0
certifi             2019.11.28
chardet             3.0.4
colorama            0.4.3
demjson             2.2.4
entrypoints         0.3
flake8              3.7.9
future              0.18.2
html5lib            1.0.1
idna                2.8
imageio             2.6.1
isort               4.3.21
lazy-object-proxy   1.4.3
mccabe              0.6.1
mechanize           0.4.4
more-itertools      8.0.2
numpy               1.17.4
packaging           19.2
pefile              2019.4.18
Pillow              6.2.1
pip                 19.3.1
pluggy              0.13.1
py                  1.8.0
py2exe              0.9.4.0
pycodestyle         2.5.0
pyflakes            2.1.1
pylint              2.4.4
pyparsing           2.4.5
PySocks             1.7.1
pytest              5.3.2
requests            2.22.0
setuptools          41.2.0
six                 1.13.0
soupsieve           1.9.5
urllib3             1.25.7
wcwidth             0.1.7
webencodings        0.5.1
wheel               0.33.6
win-unicode-console 0.5
wrapt               1.11.2
```

**Comment by sl1pkn07** [comment]
my list (python3)

~~~
└───╼  pip list
Package                        Version           
------------------------------ ------------------
acme                           1.0.0             
alabaster                      0.7.12            
anytree                        2.8.0             
apipkg                         1.5               
apng                           0.3.3             
appdirs                        1.4.3             
arrow                          0.15.5            
asciinema                      2.0.2             
asn1crypto                     1.3.0             
atomicwrites                   1.3.0             
attrs                          19.3.0            
audiotools                     3.2a2             
autobahn                       20.1.2            
Automat                        0.8.0             
Babel                          2.8.0             
backcall                       0.1.0             
Beaker                         1.11.0            
beautifulsoup4                 4.8.2             
bleach                         3.1.0             
blinker                        1.4               
bottle                         0.12.18           
breathe                        4.14.0            
brotlipy                       0.7.0             
btrfsutil                      1.1.1             
CacheControl                   0.12.6            
cachetools                     3.1.1             
certbot                        1.0.0             
certbot-dns-cloudflare         1.0.0             
certifi                        2019.11.28        
cffi                           1.13.2            
chardet                        3.0.4             
Click                          7.0               
cloudflare                     2.4.0             
colorama                       0.4.3             
ConfigArgParse                 0.15.2            
configobj                      5.0.6             
constantly                     15.1.0            
contextlib2                    0.6.0             
coverage                       4.5.4             
cryptography                   2.8               
cupshelpers                    1.0               
cycler                         0.10.0            
Cython                         0.29.14           
decorator                      4.4.1             
defusedxml                     0.6.0             
demjson                        2.2.4             
diff-match-patch               20181111          
distlib                        0.3.0             
distro                         1.4.0             
docutils                       0.14              
ds4drv                         0.5.1             
ecdsa                          0.13.3            
entrypoints                    0.3               
esptool                        2.9.dev0          
evdev                          1.2.0             
fail2ban                       0.10.5            
Flask                          1.1.1             
fusepy                         3.0.1             
future                         0.18.2            
gevent                         1.4.0             
google-api-python-client       1.7.11            
google-auth                    1.10.0            
google-auth-httplib2           0.0.3             
gps                            3.20              
graphviz                       0.13.2            
greenlet                       0.4.15            
h5py                           2.10.0            
html2text                      2019.8.11         
html5lib                       1.0.1             
httpbin                        0.7.0             
httplib2                       0.15.0            
hyperlink                      19.0.0            
idna                           2.8               
ifaddr                         0.1.6             
imageio                        2.6.1             
imagesize                      1.2.0             
importlib-metadata             0.23              
incremental                    17.5.0            
iniconfig                      1.0.0             
iotop                          0.6               
ipykernel                      5.1.3             
ipython                        7.11.1            
ipython-genutils               0.1.0             
ipywidgets                     7.5.1             
isc                            2.0               
itsdangerous                   1.1.0             
jedi                           0.15.1            
Jinja2                         2.10.3            
josepy                         1.2.0             
json5                          0.8.5             
jsonlines                      1.2.0             
jsonschema                     3.2.0             
jupyter-client                 5.3.4             
jupyter-console                6.0.0             
jupyter-core                   4.6.1             
kapidox                        5.66.0            
kiwisolver                     1.1.0             
lensfun                        0.3.2             
leveldb                        0.20              
libfdt                         1.5.0             
libvirt-python                 5.9.0             
lit                            0.9.1.dev0        
lockfile                       0.12.2            
louis                          3.12.0            
lxml                           4.4.2             
Mako                           1.1.0             
mallard-ducktype               1.0.2             
Markdown                       3.1.1             
MarkupSafe                     1.1.1             
marshmallow                    2.20.5            
matplotlib                     3.1.2             
mechanize                      0.4.5             
mercurial                      5.2.2             
meson                          0.53.0            
mistune                        0.8.4             
mock                           3.0.5             
more-itertools                 8.1.0             
mpi4py                         3.0.3             
msgpack                        0.6.2             
namcap                         3.2.10            
nbconvert                      5.6.1             
nbformat                       4.4.0             
netifaces                      0.10.9            
nftables                       0.1               
nicobatty-flasgger             0.9.4             
nine                           1.0.0             
nose                           1.3.7             
notebook                       6.0.2             
notify2                        0.3.1             
numpy                          1.18.1            
numpy-stl                      2.10.0            
oauth2client                   4.1.3             
olefile                        0.46              
ordered-set                    3.1.1             
packaging                      20.0              
pandocfilters                  1.4.2             
parsedatetime                  2.5               
parso                          0.5.2             
pbr                            5.4.4             
pep517                         0.6.0             
pexpect                        4.7.0             
pickleshare                    0.7.5             
Pillow                         6.2.1             
pip                            19.3              
pip2pkgbuild                   0.2.6             
Pivy                           0.6.5a4           
pkgconfig                      1.5.1             
platformio                     4.1.1b7           
pluggy                         0.13.1            
ply                            3.11              
Printrun                       2.0.0rc5          
progress                       1.5               
prometheus-client              0.7.1             
prompt-toolkit                 3.0.2             
protobuf                       3.11.2            
psvtools                       0.1.0             
ptyprocess                     0.6.0             
pulsectl                       19.10.4           
pwquality                      1.4.2             
py                             1.8.1             
pyaes                          1.6.1             
pyalpm                         0.9.0             
pyasn1                         0.4.8             
pyasn1-modules                 0.2.8             
pybind11                       2.4.3             
pycairo                        1.18.2            
pycparser                      2.19              
pycryptodome                   3.9.4             
pycups                         1.9.74            
pycurl                         7.43.0.3          
pydotplus                      2.0.2             
pyelftools                     0.25              
pyglet                         1.4.9             
Pygments                       2.5.2             
PyGObject                      3.34.0            
PyHamcrest                     1.9.0             
pyinotify                      0.9.6             
pylibacl                       0.5.4             
PyOpenGL                       3.1.5             
pyOpenSSL                      19.1.0            
pyparsing                      2.4.6             
Pypubsub                       4.0.3             
PyQt5                          5.14.1            
PyQt5-sip                      12.7.0            
pyRFC3339                      1.1               
pyrsistent                     0.15.7            
pyserial                       3.4               
PySocks                        1.7.0             
pytest                         5.3.2             
pytest-httpbin                 1.0.0             
pytest-mock                    2.0.0             
pytest-runner                  5.2               
python-dateutil                2.8.1             
python-gflags                  3.1.2             
python-utils                   2.3.0             
pytoml                         0.1.21            
pytz                           2019.3            
pyudev                         0.21.0.dev20191102
pyxdg                          0.26              
PyYAML                         5.1.2             
pyzmq                          18.1.1            
QtPy                           1.9.0             
raven                          6.10.0            
Reflector                      2019.10.15.3.19.34
requests                       2.22.0            
requests-toolbelt              0.9.1             
retrying                       1.3.3             
rpm                            4.15.1            
rsa                            4.0               
scipy                          1.4.1             
scons                          3.1.2             
semantic-version               2.8.2             
Send2Trash                     1.5.0             
service-identity               18.1.0            
setuptools                     45.0.0            
Shapely                        1.6.4.post2       
simplejson                     3.17.0            
sip                            4.19.20           
six                            1.13.0            
smbus                          1.1               
snowballstemmer                2.0.0             
soupsieve                      1.9.5             
Sphinx                         2.2.1             
sphinx-bootstrap-theme         0.7.1             
sphinx-rtd-theme               0.4.3             
sphinxcontrib-applehelp        1.0.1             
sphinxcontrib-devhelp          1.0.1             
sphinxcontrib-htmlhelp         1.0.2             
sphinxcontrib-jsmath           1.0.1             
sphinxcontrib-qthelp           1.0.2             
sphinxcontrib-serializinghtml  1.1.3             
sphinxcontrib-svg2pdfconverter 0.1.0             
sphinxcontrib-websupport       1.1.2             
SQLAlchemy                     1.3.12            
systemd-python                 234               
tabulate                       0.8.6             
terminado                      0.8.3             
testpath                       0.4.4             
tornado                        6.0.3             
traitlets                      4.3.3             
translate-toolkit              2.4.0             
trimesh                        3.4.0             
Twisted                        19.10.0           
txaio                          18.8.1            
uritemplate                    3.0.1             
urllib3                        1.25.7            
urwid                          2.1.0             
virtualenv                     16.1.0            
wcwidth                        0.1.8             
webencodings                   0.5.1             
Werkzeug                       0.16.0            
Whoosh                         2.7.4             
widgetsnbextension             3.5.1             
wsaccel                        0.6.2             
wxPython                       4.0.7.post2       
xattr                          0.9.7             
youtube-dl                     2020.1.15         
zeroconf                       0.23.0            
zipp                           1.0.0             
zope.component                 4.6               
zope.deferredimport            4.3.1             
zope.deprecation               4.4.0             
zope.event                     4.4               
zope.hookable                  5.0.0             
zope.interface                 4.7.1             
zope.proxy                     4.3.3 
~~~
 but none of them is installed througth pip, only for package manager

**Comment by Nandaka** [comment]
not sure if there are some changes from 1.7.0 to 1.7.1

**Comment by bluerthanever** [comment]
Actually I ran into this error this morning and I had been looking at it for a while....
For me it looks like the cookie string "PHPSSID" expired, that PixivUtil2 keeps trying to visit https://www.pixiv.net/ and the "return_to=/", and then "return_to=/return_to=/" or something like this, while the server keeps telling PixivUtil2 that the "PHPSSID" is "deleted".
I just tried to log in with browser and copy then paste the string into config.ini, since I later found out that I can't log in with PixivUtil2 any more that it requires some captcha string.
[https://github.com/Nandaka/PixivUtil2/issues/541](url)

**Comment by sl1pkn07** [comment]
fixed with https://github.com/Nandaka/PixivUtil2/issues/505#issuecomment-510539272

thanks @bluerthanever for the tip

greetings

**Comment by bluerthanever** [comment]
> 
> 
> fixed with [#505 (comment)](https://github.com/Nandaka/PixivUtil2/issues/505#issuecomment-510539272)
> 
> thanks @bluerthanever for the tip
> 
> greetings

Glad that I can help.
